api/v1alpha3/awscluster_types.go

```diff
@@ -82,6 +82,23 @@ type AWSClusterSpec struct {
 	// Bastion contains options to configure the bastion host.
 	// +optional
 	Bastion Bastion `json:"bastion"`
+
+	// Endpoints contains options to map custom AWS endpoints to services.
+	// +optional
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
```

Comment:
Would you consider endpoints to be distinct from credentials in that the same set of credentials would be used across multiple sets of endpoints? 

I'm wondering if it would make sense to tie the endpoints more closely with the credentials used (which we currently configured at the controller-manager level), but there is an outstanding issue to add multitenancy support to a single controller instance here: https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1552

@randomvariable thoughts?

Reply:
That makes sense; I built the schema around what AWS presents in its SDK.